KesaApplication/ViewController.swift

```diff
@@ -8,18 +8,180 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, UITextFieldDelegate {
 
+    
+    @IBOutlet weak var firstNameLabel: UILabel!
+    @IBOutlet weak var lastNameLabel: UILabel!
+    @IBOutlet weak var emailLabel: UILabel!
+    @IBOutlet weak var passwordLabel: UILabel!
+    
+    
+    @IBOutlet weak var firstName: UITextField!
+    @IBOutlet weak var lastName: UITextField!
+    @IBOutlet weak var email: UITextField!
+    @IBOutlet weak var password: UITextField!
+    
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+        
+        firstName.delegate = self // set delegate to 4 text fields
+        lastName.delegate = self
+        email.delegate = self
+        password.delegate = self
+        
     }
-
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    
+    @IBAction func onLoginButtonClickListener(sender: UIButton) {
+        
+        /* Once login button is pressed, text fields are no longer important */
+        self.firstName.resignFirstResponder() // Give up first responder status
+        self.lastName.resignFirstResponder()
+        self.email.resignFirstResponder()
+        self.password.resignFirstResponder()
+    }
+    
+    
+    /* When the screen is touched */
+    override func touchesBegan(touches: Set<UITouch>,
+                               withEvent event: UIEvent?) {
+        self.view.endEditing(true) /* For instances where the keyboard is 
+                                      already brought up, Hide the keyboard */
+        firstNameLabel.textColor = UIColor.blackColor()
+        lastNameLabel.textColor = UIColor.blackColor()
+        emailLabel.textColor = UIColor.blackColor()
+        passwordLabel.textColor = UIColor.blackColor()
     }
+    
+    
+    
+    // Highlighting each labels when their text fields are chosen
+    func textFieldDidBeginEditing(textField: UITextField) {
```

Comment:
Hmm.. this is not really better than the last implementation you had.
Please consider changing the implementation to something like this.

``` swift
 firstNameLabel.textColor = UIColor.blackColor()
 lastNameLabel.textColor = UIColor.blackColor()
 emailLabel.textColor = UIColor.blackColor()
 passwordLabel.textColor = UIColor.blackColor()

switch textField {
 case ...
 case ...
 case ...
}

```